app.py

```diff
@@ -0,0 +1,429 @@
+import streamlit as st
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import pickle
+import seaborn as sns
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.model_selection import RandomizedSearchCV, GridSearchCV
+from sklearn.metrics import r2_score
+from sklearn.metrics import mean_squared_error
+from sklearn.metrics import mean_absolute_error
+from mpl_toolkits.mplot3d import Axes3D
+from PIL import Image
+from plotly.offline import iplot
+import plotly as py
+import plotly.tools as tls
+import cufflinks as cf
+import plotly.figure_factory as ff
+import plotly.express as px
+import streamlit.components.v1 as components
+
+
+py.offline.init_notebook_mode(connected = True)
+cf.go_offline()
+cf.set_config_file(theme='white')
+
+df = pd.read_csv("forestfires.csv")
+#all functions
+# @st.cache
+def welcome():
+
+    return "Welcome All"
+# @st.cache    
+def predict_forest_fire(x,y,month,ffmc,dmc,dc,temp):
+    prediction = model.predict([[x,y,month,ffmc,dmc,dc,temp]])
+    print(prediction)
+    return prediction
+
+# @st.cache
+def main_prediction():
+    # st.title("Forest Fire Prediction")
+    # st.markdown("""<center><img src='https://upload.wikimedia.org/wikipedia/commons/d/d8/Deerfire_high_res_edit.jpg' width='500px'></center>""",unsafe_allow_html=True)
+    # st.markdown("""<br>""",unsafe_allow_html=True)
+    
+    html_temp = """
+    <div style="background-color:#00b3ff;padding:10px">
+    <h2 style="color:white;text-align:center;">Forest Fire Prediction</h2>
+    </div>
+    """
+    st.markdown(html_temp,unsafe_allow_html=True)
+    st.markdown("""<br>""",unsafe_allow_html=True)
+    x = st.text_input("X")
+    y = st.text_input("Y")
+    month = st.text_input("Month")
+    ffmc = st.text_input("FFMC")
+    dmc = st.text_input("DMC")
+    dc = st.text_input("DC")
+    temp = st.text_input("Temp")
+    result=""
+    if st.button("Predict"):
+        try:
+            result=predict_forest_fire(x,y,month,ffmc,dmc,dc,temp)
+            st.success('The output is {}'.format(result))
+            st.balloons()
+            
+        except:
+            st.error("Please fill up all the inputs🙏")
```

Comment:
Make the exception handling a bit more robust. `Please fill up all the inputs` will still show even if the user has entered all the fields. Add different message for different Exceptions without printing the entire stacktrace.

Reply:
this is what I am talking about: 
![image](https://user-images.githubusercontent.com/29832615/97302439-18bec200-187f-11eb-9206-e49d411c9039.png)

Reply:
Actually, all the fields accept only numeric value. so, here `Jan` must be a number.
Yes, I can write the instruction to fill the inputs with only numeric values.

Reply:
Yeah that's I am asking you to make the exceptions more clearer to understand.

Reply:
I have fixed the exception with a proper massage. So, now if anybody enters any non-numeric value or doesn't enter any value in the inputs then they will see a message.
<img src="https://i.imgur.com/0DTOhfX.png">